From 2d5a3a2fff0603c95ddb86707ddde5fc823146ea Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 20 Jan 2022 04:53:33 +0200
Subject: [PATCH] [SW-69190] gaudi3: update addr space in linker

Update linker with ddresses for code and data as per address space
for gaudi3.

Change-Id: I9a3426eceebceee3d3ae92cafc7ee33ac4a514f8
---

diff --git a/arch/gaudi3/arc/arcboot_rom.lds b/arch/gaudi3/arc/arcboot_rom.lds
index ece1742..97fc414 100644
--- a/arch/gaudi3/arc/arcboot_rom.lds
+++ b/arch/gaudi3/arc/arcboot_rom.lds
@@ -6,17 +6,16 @@
 OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
 OUTPUT_ARCH(arc)
 
-/* TODO: Replace with gaudi3 values */
-
 MEMORY
 {
-	ICCM	: ORIGIN = 0x00000000, LENGTH = 128K
-	STM	: ORIGIN = 0x24000000, LENGTH = 32M
-	SPI	: ORIGIN = 0x26000000, LENGTH = 16M
-	SBROM	: ORIGIN = 0x27FC0000, LENGTH = 64K
-	FWSRAM	: ORIGIN = 0x27FD0000, LENGTH = 64K
-	CS	: ORIGIN = 0x2C000000, LENGTH = 32M
-	DCCM	: ORIGIN = 0x80000000, LENGTH = 32K
+	ICCM	: ORIGIN = 0x00000000, LENGTH = 256K
+	STM	: ORIGIN = 0x20000000, LENGTH = 32M
+	SPI	: ORIGIN = 0x22000000, LENGTH = 16M
+	SBROM	: ORIGIN = 0x23100000, LENGTH = 64K
+	SPSRAM	: ORIGIN = 0x23120000, LENGTH = 64K
+	FWSRAM	: ORIGIN = 0x23140000, LENGTH = 128K
+	DCCM	: ORIGIN = 0x80000000, LENGTH = 64K
+	CS	: ORIGIN = 0xF0000000, LENGTH = 32M
 }
 
 REGION_ALIAS("REGION_IVT",	SBROM)
diff --git a/arch/gaudi3/arc/onix_arcboot_rom.lds b/arch/gaudi3/arc/onix_arcboot_rom.lds
index 8564bf3..a013c5c 100644
--- a/arch/gaudi3/arc/onix_arcboot_rom.lds
+++ b/arch/gaudi3/arc/onix_arcboot_rom.lds
@@ -8,5 +8,6 @@
 /* This is SRAM address, used for Onix targets */
 MEMORY
 {
+	/* TODO: Update address to run in onix env */
 	SBROM	: ORIGIN = 0x11000000, LENGTH = 64K
 }
